src/frontends/tensorflow/src/graph_iterator_saved_model.hpp

```diff
@@ -0,0 +1,185 @@
+    void readBundleHeader();
+    void readCMOGraph();
+};
+
+// Loads graph from Tensorflow Saved Model file (saved_model.pb)
+class GraphIteratorSavedModel : public GraphIteratorProto {
+    std::shared_ptr<::tensorflow::SavedModel> m_saved_model;
+    std::shared_ptr<SMVariablesIndex> m_variables_index;
+
+public:
+    template <typename T>
+    GraphIteratorSavedModel(const std::basic_string<T>& path)
+        : m_saved_model(std::make_shared<::tensorflow::SavedModel>()) {
+        this->readSavedModel(path);
+    }
+
+    static bool isSavedModel(const std::string& path);
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+    static bool isSavedModel(const std::wstring& path);
+#endif
+
+    std::shared_ptr<SMVariablesIndex> getVariablesIndex() {
+        return m_variables_index;
+    }
+
+private:
+    bool isValidSignature(const ::tensorflow::SignatureDef& signature);
+
+    template <typename T>
+    bool readSavedModel(const std::basic_string<T>& path) {
+        std::ifstream sm_stream{path + getSMName<T>(), std::ifstream::in | std::ifstream::binary};
+        FRONT_END_GENERAL_CHECK(sm_stream && sm_stream.is_open(), "Model file does not exist");
+
+        std::basic_string<T> varIndexPath = path + getVIName<T>();
+        if (ov::util::file_exists(varIndexPath)) {
+            m_variables_index = std::make_shared<SMVariablesIndex>();
+            std::ifstream vi_stream{varIndexPath, std::ifstream::in | std::ifstream::binary};
+            FRONT_END_GENERAL_CHECK(vi_stream && vi_stream.is_open(),
+                                    "Saved Model's variable index file does not exist");
+            FRONT_END_GENERAL_CHECK(m_variables_index->readVariables(vi_stream, path),
+                                    "Saved Model's variable index file cannot be parsed");
+        }
+
+        bool res = m_saved_model->ParseFromIstream(&sm_stream);
+        FRONT_END_GENERAL_CHECK(res && m_saved_model->meta_graphs_size(), "Saved Model cannot be parsed");
+
+        // Supported only first meta_graph at the moment
+        const ::tensorflow::MetaGraphDef& meta_graph = m_saved_model->meta_graphs(0);
+        FRONT_END_GENERAL_CHECK(meta_graph.has_graph_def(), "Saved Model doesn't contain GraphDef");
+
+        std::vector<std::string> validSignatures = {};
+        for (auto& sit : meta_graph.signature_def()) {
+            const std::string& key = sit.first;
+            const ::tensorflow::SignatureDef& val = sit.second;
+            if (isValidSignature(val)) {
+                validSignatures.push_back(key);
+            } else {
+                OPENVINO_ASSERT(false, "Saved Model contains invalid signatures");
+            }
+        }
+
+        // TODO: assets reading
+
+        m_graph_def = std::make_shared<::tensorflow::GraphDef>(meta_graph.graph_def());
+
+        // TODO: update loading nodes by using data from separately
```

Comment:
please fix this or create a ticket

Reply:
Sure. This activity is in progress with restoring variables values. I’ll remove it when it’ll be finished. Right now it is one of places I’m trying change to handle unsupported strings.